src/main/java/com/conveyal/r5/streets/Split.java

```diff
@@ -98,16 +98,31 @@ public static Split find (double lat, double lon, double searchRadiusMeters,
 
             curr.edge = e;
             edge.seek(e);
+            //LOG.info("Considering OSM way {}", edge.getOSMID());
             //Skip Link edges those are links between transit stops/P+R/Bike share vertices and graph
             //Without this origin or destination point can link to those edges because they have ALL permissions
             //and route is never found since point is inaccessible because edges leading to it don't have required permission
             if (edge.getFlag(EdgeStore.EdgeFlag.LINK)) return true;
 
-            // FIXME this is only checking the forward edge permissions, even though we're iterating over and linking to edge _pairs_
-            // If an edge does not allow traversal with the specified mode, skip over it.
-            if (streetMode == StreetMode.WALK && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_PEDESTRIAN)) return true;
-            if (streetMode == StreetMode.BICYCLE && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_BIKE)) return true;
-            if (streetMode == StreetMode.CAR && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_CAR)) return true;
+            if (!edge.allowsStreetMode(streetMode)) {
+                //LOG.info("rejecting edge {} (OSM way {} forward)", edge.edgeIndex, edge.getOSMID());
+                return true;
+            }
+            edge.advance();
+            if (!edge.allowsStreetMode(streetMode)) {
+                //LOG.info("rejecting edge {} (OSM way {} backward)", edge.edgeIndex, edge.getOSMID());
+                return true;
+            }
+            edge.retreat();
+
+//            if (!edge.allowsStreetMode(streetMode)) {
+//                // The edge does not allow forward traversal with the specified mode, try the backward edge.
+//                edge.advance();
+//                // If backward traversal is also not allowed, skip this edge and try the next one.
+//                if (!edge.allowsStreetMode(streetMode)) return true;
+//            }
+            // FIXME we are splitting only one direction, we actually want to split the two directions separately, so either of distance 1 or distance 2 can be infinite.
+            // FIXME we should probably entirely skip over motorways and link only surface streets.
```

Comment:
Using the `LINKABLE` flag?

Reply:
`LINKABLE` seems to be currently used just for linking park-and-ride lots, but yes I'd say its use should be extended to linking origins and destinations. The set of tags checked seems correct: `if (!(way.hasTag("tunnel", "yes") || way.hasTag("covered", "yes") || way.hasTag("highway", "motorway")))`. Let's make another ticket for this because we need to test the results thoroughly before releasing such a change.